File: src/com/cattailsw/nanidroid/ShioriProtocolVersion.java
package com.cattailsw.nanidroid;

/** Immutable SHIORI wire-protocol version; replaces Apache's HTTP-only type. */
public final class ShioriProtocolVersion {
    private final String protocol;
    private final int major;
    private final int minor;

    public ShioriProtocolVersion(String protocol, int major, int minor) {
        this.protocol = protocol;
        this.major = major;
        this.minor = minor;
    }

    @Override
    public String toString() {
        return protocol + "/" + major + "." + minor;
    }
}

File: src/main/kotlin/com/cattailsw/nanidroid/ShioriProtocolVersion.kt
package com.cattailsw.nanidroid

/** Immutable SHIORI wire-protocol version; replaces Apache's HTTP-only type. */
class ShioriProtocolVersion(
    private val protocol: String,
    private val major: Int,
    private val minor: Int,
) {
    override fun toString(): String = "$protocol/$major.$minor"
}
